New-HostProcessBaseImage.ps1

```diff
@@ -0,0 +1,140 @@
+"1.0" | Out-File -FilePath "build\image\${layerHash}\VERSION" -Encoding ascii
+Copy-Item -Path  "build\layer\layer.tar" -Destination "build\image\${layerHash}\layer.tar"
+
+$now = [DateTime]::UtcNow.ToString("o")
+@"
+{
+    "id": "${layerHash}",
+    "created": "${now}",
+    "container_config": {
+        "Hostname": "",
+        "Domainname": "",
+        "User": "",
+        "AttachStdin": false,
+        "AttachStdout": false,
+        "AttachStderr": false,
+        "Tty": false,
+        "OpenStdin": false,
+        "StdinOnce": false,
+        "Env": null,
+        "Cmd": null,
+        "Image": "",
+        "Volumes": null,
+        "WorkingDir": "",
+        "Entrypoint": null,
+        "OnBuild": null,
+        "Labels": null
+    },
+    "config": {
+        "Hostname": "",
+        "Domainname": "",
+        "User": "ContainerUser",
+        "AttachStdin": false,
+        "AttachStdout": false,
+        "AttachStderr": false,
+        "Tty": false,
+        "OpenStdin": false,
+        "StdinOnce": false,
+        "Env": null,
+        "Cmd": [
+            "c:\\windows\\system32\\cmd.exe"
+        ],
+        "Image": "",
+        "Volumes": null,
+        "WorkingDir": "",
+        "Entrypoint": null,
+        "OnBuild": null,
+        "Labels": null
+    },
+    "architecture": "amd64",
+    "os": "windows"
+}
+"@ | Out-File -FilePath "build\image\${layerHash}\json" -Encoding ascii
+
+
+# Create the image config and manifest files
+@"
+{
+    "architecture": "amd64",
+    "config": {
+        "Hostname": "",
+        "Domainname": "",
+        "User": "",
+        "AttachStdin": false,
+        "AttachStdout": false,
+        "AttachStderr": false,
+        "Tty": false,
+        "OpenStdin": false,
+        "StdinOnce": false,
+        "Env": null,
+        "Cmd": [
+            "c:\\windows\\system32\\cmd.exe"
+        ],
+        "Image": "",
+        "Volumes": null,
+        "WorkingDir": "",
+        "Entrypoint": null,
+        "OnBuild": null,
+        "Labels": null
+    },
+    "created": "${now}",
+    "history": [
+        {
+            "created": "${now}"
+        }
+    ],
+    "os": "windows",
+    "os.version": "10.0.17763.1",
+    "rootfs": {
+        "type": "layers",
+        "diff_ids": [
+            "sha256:${layerHash}"
+        ]
+    }
+}
+"@ | Out-File -FilePath "build\image\config.json" -Encoding ascii
+$configHash = (Get-FileHash -Algorithm SHA256 "build\image\config.json").Hash.ToLower()
+Move-Item -Path "build\image\config.json" -Destination "build\image\${configHash}.json"
+
+@"
+[
+    {
+        "Config": "${configHash}.json",
+        "Layers": [
+            "${layerHash}/layer.tar"
+        ]
+    }
+]
+"@ | Out-File  -FilePath "build\image\manifest.json" -Encoding ascii
+
+# Tar the image
+tar.exe  -cf "build\host-process-scratch.tar" -C "build\image" .
+
+# Output a file with the image hash so we can import/push the image from CI
```

Comment:
looks like the outcome of this script is tar file and manifest files? do we have anything that pushes it to a registry now?

Reply:
No. 
The GH action in my repo where this is based on does a docker load + push.
We can't use GH actions to push to mcr.microsoft.com so I intentionally left that out.
I can document the steps we use to load + push in a follow up PR (once we've got the image published)

Reply:
x-ref https://github.com/marosset/host-process-scratch-image/blob/94c662a6a7615d5a2fea1b6d22948df5d0c69010/.github/workflows/build.yaml#L30-L37

Reply:
sounds good!